File: src/dynamicProgramming/AntWarrior.java
package dynamicProgramming;

import java.io.*;

public class AntWarrior {
    public static void main(String[] args) throws IOException {
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int N = Integer.parseInt(br.readLine());
        String[] inputs = br.readLine().split(" ");

        int[] food = new int[N];

        for (int i = 0; i < N; i++) {
            food[i] = Integer.parseInt(inputs[i]);
        }

        int[] dp = new int[N];

        dp[0] = food[0];
        dp[1] = Math.max(food[0], food[1]);

        for (int i = 2; i < N; i++) {
            dp[i] = Math.max(dp[i - 1], dp[i - 2] + food[i]);
        }

        bw.write(String.valueOf(dp[N-1]));
        bw.flush();
        bw.close();
    }
}

File: JS/dynamicProgramming/antWarrior.js
"use strict";

const fs = require("fs");

const input =
  process.platform === "linux"
    ? fs.readFileSync("/dev/stdin").toString().split("\n")
    : `4
1 3 1 5`.split("\n");

const N = parseInt(input[0]);
const food = input[1].split(" ").map((i) => parseInt(i));
let dp = Array(N).fill(0);

dp[0] = food[0];
dp[1] = Math.max(food[0], food[1]);
for (let i = 2; i < N; i++) {
  dp[i] = Math.max(dp[i - 1], dp[i - 2] + food[i]);
}

console.log(dp[N - 1]);
